exhibitor-core/src/main/java/com/netflix/exhibitor/core/azure/PropertyBasedAzureCredential.java

```diff
@@ -0,0 +1,51 @@
+package com.netflix.exhibitor.core.azure;
+
+import org.apache.curator.utils.CloseableUtils;
+
+import java.io.*;
+import java.util.Properties;
+
+public class PropertyBasedAzureCredential implements AzureCredential {
+    private final String accountName;
+    private final String accountKey;
+
+    public static final String PROPERTY_AZURE_ACCOUNT_NAME = "com.netflix.exhibitor.azure.account-name";
+    public static final String PROPERTY_AZURE_ACCOUNT_KEY = "com.netflix.exhibitor.azure.account-key";
+
+    public PropertyBasedAzureCredential(File propertiesFile) throws IOException
```

Comment:
I really don't like constructors that throw - it's an Anti-Pattern.

I'd rather catch the `IOException`, mark the class as invalid or something, or maybe add a static factory  method (that can throw).

In fact, looking at the other constructor, I'd convert this one into a Factory Method:

    static public AzureCredential loadPropertiesBasedCredentials(File properties) {
        try {
            Properties props = loadProperties(properties);
            return new PropertyBasedAzureCredential(props);
        } catch (IOException ex) {
              // do the needful here
        }
    }

or something similar.